test/joke_test.rb

```diff
@@ -3,9 +3,26 @@
 require './lib/joke'
 
 class JokeTest < Minitest::Test
+
+  def setup
+    @joke = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+  end
+
+
   def test_it_exists
+    assert_instance_of Joke, @joke
   end
 
   def test_it_has_attributes
+    assert_equal 1, @joke.id
+  end
+
+  def test_joke_setup
+    assert_equal "Why did the strawberry cross the road?", @joke.setup
+  end
+
+  def test_joke_punchline
+    assert_equal "Because his mother was in a jam.", @joke.punchline
   end
```

Comment:
Good testing in here. It's okay to group into a single test any attributes that get set within the `initialize` method.